text/dialect_schema.yaml

```diff
@@ -0,0 +1,203 @@
+      required_options:
+        type: object
+        description: >
+          Function-specific options that must be supported by the dialect.  These
+          are the exact list of options that must be used.  If no options are
+          listed here, no options must be present.
+      supported_impls:
+        type: array
+        description: >
+          List of implementations (parameter type lists) supported by this function.
+          For example, i32 or str_str.  Variadic arguments are not included in
+          the impl's list of parameters.
+        minItems: 1
+        items:
+          type: string
+      variadic:
+        type: integer
+        description: >
+          If present, indicates the function accepts a variable number of arguments
+          The value specifies the minimum number of arguments required (must be ≥ 1).
+        minimum: 1
+  dialect_relation:
+    type: object
+    additionalProperties: false
+    required: [rel_type]
+    properties:
+      rel_type:
+        type: string
+        description: >
+          rel_type corresponds to the Rel rel_type oneof field names (in uppercase).
+          Only relation types specified here are supported in this dialect.  If a
+          relation is listed here, all features of that relation must be supported
+          for the dialect's requested version of Substrait (in other words if
+          there is a deprecated feature at that Substrait version it does not need
+          to be supported).
+        enum:
+          - READ
+          - FILTER
+          - FETCH
+          - AGGREGATE
+          - SORT
+          - JOIN
+          - PROJECT
+          - SET
+          - EXTENSION_SINGLE
+          - EXTENSION_MULTI
+          - EXTENSION_LEAD
+          - CROSS
+          - REFERENCE
+          - WRITE
+          - DDL
+          - UPDATE
+          - HASH_JOIN
+          - MERGE_JOIN
+          - NESTED_LOOP_JOIN
+          - WINDOW
+          - EXCHANGE
+          - EXPAND
+      sub_type:
+        type: array
+        description: >
+          sub_type corresponds to the primary oneof in any given relation type.
+          For example, in ReadRel it is the field name inside read_type and
+          in JoinRel it is the join_type.  If there is no oneof in a relation
+          sub_type should be empty.  Functionality not controlled by the
+          sub_type needs to be supported for the relation to be included in the
+          dialect.  For user-defined types, use the URL that describes the type
+          of the message (for example type.googleapis.com/google.profile.Person).
+        items:
+          type: string
+  dialect_expression:
+    type: object
+    additionalProperties: false
+    required: [rex_type]
+    properties:
+      rex_type:
+        type: string
+        description: rex_type corresponds to the Expression rex_type oneof field names.
+        enum:
+          - LITERAL
```

Comment:
same comment as relations about strings versus objects. I may support only certain subquery type for example.